pkg/chain/chain.go

```diff
@@ -30,135 +86,162 @@ type Handle interface {
 	// In case the block is not yet mined, an error should be returned.
 	BlockTimestamp(blockNumber *big.Int) (uint64, error)
 
-	BondedECDSAKeepFactory
-	BondedECDSAKeep
+	PublicKeyToOperatorID(publicKey *cecdsa.PublicKey) OperatorID
+
+	// BondedECDSAKeepManager returns a handle to the on-chain component used to
+	// interact with bonded ECDSA keeps.
+	BondedECDSAKeepManager() (BondedECDSAKeepManager, error)
 }
 
-// BondedECDSAKeepFactory is an interface that provides ability to interact with
-// BondedECDSAKeepFactory ethereum contracts.
-type BondedECDSAKeepFactory interface {
-	// RegisterAsMemberCandidate registers client as a candidate to be selected
-	// to a keep.
-	RegisterAsMemberCandidate(application common.Address) error
+// BondedECDSAKeepManager is an interface whose implementations allow for
+// managing, monitoring, and creating bonded ECDSA keeps as well as the
+// operator's eligibility for participation in such keeps.
+type BondedECDSAKeepManager interface {
+	// TBTCApplicationHandle returns a handle for interacting with the tBTC
+	// application associated with this BondedECDSAKeepManager. Returns nil with
+	// an error if no tBTC application exists for this manager.
+	TBTCApplicationHandle() (TBTCHandle, error)
```

Comment:
I could see it going either way. The tBTC application is a BondedECDSAKeep application. I originally played with having this return a `BondedECDSAKeepApplicationHandle`, but realized that just added complexity to consumers (suddenly you have to instantiate the tBTC stuff separately with the same stack of chain dependencies) and the gain in flexibility wasn't really clear to me. `TBTCHandle` embeds `BondedECDSAKeepApplicationHandle`, so any functions that should be generic across applications can be, while someone who is looking specifically for this application can get the specific information.

This also allows easier iteration on a new chain version, because the methods all descend from the `chain.Handle` interface's methods, so if `chain.Handle` is implemented you know you're good to go. That wasn't a formal requirement, it just sort of fell out informally as I went through the code.

Reply:
At first, I wasn't into defining the application-specific handle here, I was thinking about something more generic. But after considering the complexity of being generic here it sounds fine to be tBTC specific.